openwisp_controller/pki/api/views.py

```diff
@@ -49,6 +52,21 @@ class CaDetailView(ProtectedAPIMixin, RetrieveUpdateDestroyAPIView):
     queryset = Ca.objects.all()
 
 
+class CaRenewView(ProtectedAPIMixin, GenericAPIView):
+    serializer_class = serializers.Serializer
+    queryset = Ca.objects.all()
+
+    def post(self, request, pk):
+        """
+        Renews the CA.
+        """
+        instance = self.get_object()
+        instance.renew()
+        return Response(
+            {_("CA '{}' renewed successfully".format(instance.name))}, status=302,
+        )
```

Comment:
return the serialized data with the updated data and 200 as status code, applies to the other endpoints as well

Reply:
This is done. :+1: